### PR TITLE
Added wrapper and build process for docker container

### PR DESCRIPTION
**Pull request by JakeNarkizian**
Also some minor fixes in separate commits. 

jenkins, skip this please.

### CODE REVIEW

**Comment by JakeNarkizian** [comment]
@hannes-ucsc please take a look.

File: Makefile [hunk 1]
```diff
@@ -59,15 +72,49 @@ clean_develop: check_venv
...
+docker/builds/%: build_path=docker/builds/$(protocol_ver)
+docker/builds/%: docker/Dockerfile.py docker/wrapper.py sdist
+	mkdir -p $(build_path)
+	cp docker/wrapper.py $(build_path)/
+	cp dist/$(sdist_name) $(build_path)/
+	$(python) docker/Dockerfile.py --docker-version $(true_ver) > $(build_path)/Dockerfile
+	cd $(build_path) && docker build --tag $(quay_path):$(long_commit_tag) .
+	docker tag $(quay_path):$(long_commit_tag) $(quay_path):$(short_commit_tag)
+	docker tag $(quay_path):$(long_commit_tag) $(quay_path):$(pipeline_tag)
+	docker tag $(quay_path):$(long_commit_tag) $(quay_path):$(protocol_ver)
```
Comment (hannes-ucsc) [inline Makefile]: Consider

```
for tag in $(short_commit_tag) $(pipeline_tag) $(protocol_ver); do \
    docker tag $(quay_path):$(long_commit_tag) $(quay_path):$$tag; \
done
```

File: Makefile [hunk 2]
```diff
@@ -59,15 +72,49 @@ clean_develop: check_venv
...
+clean_docker:
+	- rm -r docker/builds
+
+docker_push: $(foreach ver,$(docker_versions),docker_push_$(ver))
+docker_push_%: docker/builds/%
+	docker push $(quay_path):$(long_commit_tag)
+	docker push $(quay_path):$(short_commit_tag)
+	docker push $(quay_path):$(pipeline_tag)
+	docker push $(quay_path):$(protocol_ver)
```
Comment (hannes-ucsc) [inline Makefile]: See above

Comment (hannes-ucsc) [inline Makefile]: Or maybe be `docker push` can handle multiple tags in one command?

File: docker/Dockerfile.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                         'apt-transport-https',
+                         'ca-certificates'))
+docker_version = args.docker_version
+print textwrap.dedent("""
+    FROM ubuntu:14.04
+
+    RUN apt-get update \
+        && apt-get install -y {dependencies}
+    RUN wget https://get.docker.com/builds/Linux/x86_64/docker-{docker_version}.tgz \
+            -O /usr/local/bin/docker \
```
Comment (hannes-ucsc) [inline docker/Dockerfile.py]: Indent to align `-` with `h`

File: docker/Dockerfile.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+dependencies = ' '.join(('git',
+                         'python-dev',
+                         'python-pip',
+                         'wget',
+                         'curl',
+                         'apt-transport-https',
+                         'ca-certificates'))
+docker_version = args.docker_version
+print textwrap.dedent("""
+    FROM ubuntu:14.04
```
Comment (hannes-ucsc) [inline docker/Dockerfile.py]: We should look at basing this image on the Toil appliance. Can you open a ticket?

File: docker/Dockerfile.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    RUN apt-get update \
+        && apt-get install -y {dependencies}
+    RUN wget https://get.docker.com/builds/Linux/x86_64/docker-{docker_version}.tgz \
+            -O /usr/local/bin/docker \
+        && chmod u+x /usr/local/bin/docker
+    RUN pip install setuptools --upgrade
+    RUN pip install toil==3.5.0a1.dev274
+
+    COPY wrapper.py /opt/pipeline/
+    COPY toil-rnaseq* /opt/pipeline/sdist.tar.gz
```
Comment (hannes-ucsc) [inline docker/Dockerfile.py]: Is this supposed to copy the sdist? If so, I would parameterize the `Dockerfile` with the exact name of the `sdist` and use that here. Toil does it that way. The wildcard could bite you if someone switched a working copy between two different versions and didn't clean in between. In that case there will be two different tarballs, both matching the wildcard.

**Comment by jvivian** [comment]
@JakeNarkizian — I'm guessing I can close this?

**Comment by cket** [comment]
Let's make sure we keep the branch around at least, it will probably be useful in the future

**Comment by jvivian** [comment]
> Let's make sure we keep the branch around at least, it will probably be useful in the future

Sounds good.

**Pull request closed without merging**